### PR TITLE
[tempest] allow set of dns servers for tempest pod

### PR DESCRIPTION
**Pull request by stuggi**
As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running
- [x] Appropriate documentation exists and/or is up-to-date:
  - [x] README in the role
  - [x] Content of the docs/source is reflecting the changes

### CODE REVIEW

**Review by arxcruz — APPROVED** [review]
/lgtm

File: ci_framework/roles/tempest/tasks/main.yml [hunk 1]
```diff
@@ -56,6 +56,7 @@
     volume:
       - "{{ cifmw_tempest_artifacts_basedir }}/:/var/lib/tempest/external_files:Z"
     detach: false
+    dns: "{{ cifmw_tempest_dns_servers }}"
```
Comment (SeanMooney) [inline ci_framework/roles/tempest/tasks/main.yml:59]: we can see this being passed here

https://review.rdoproject.org/zuul/build/03fec18dcb014e0b84720314881c33bf/log/job-output.txt#5278

2023-10-10 08:25:25.007832 | controller | TASK [tempest : Run tempest name=tempest, image={{ cifmw_tempest_image }}:{{ cifmw_tempest_image_tag }}, state=started, auto_remove={{ cifmw_tempest_remove_container | default(false) }}, network=host, volume=['{{ cifmw_tempest_artifacts_basedir }}/:/var/lib/tempest/external_files:Z'], detach=False, dns={{ cifmw_tempest_dns_servers }}, env={'CONCURRENCY': '{{ cifmw_tempest_concurrency | default(omit) }}'}] ***

**Review by SeanMooney — COMMENTED** [review]
This does appear to be functional.

the implication here is we are using the DNS server provided by crc/openshift instead.

**Comment by cjeanner** [comment]
/approve

Wondering if we may compute that IP (later) in order to ensure it's the right one. Probably at zuul job boostrap level though.

**Pull request merged**